### PR TITLE
Add "mean absolute error" to metrics

### PR DESCRIPTION
**Pull request by zhengruifeng**
Just add a new regression metric "mean absolute error".

### CODE REVIEW

File: src/common/math.h [hunk 1]
```diff
@@ -24,6 +24,15 @@ inline float Sigmoid(float x) {
 }
 
 /*!
+ * \brief calculate the absolute value of the input.
+ * \param x input parameter
+ * \return the transformed value.
+ */
+inline float Abs(float x) {
```
Comment (tqchen) [inline src/common/math.h]: can use std::abs

**Comment by tqchen** [comment]
added some comment, please update the code and combine your commits https://xgboost.readthedocs.org/en/latest/dev-guide/contribute.html#how-to-combine-multiple-commits-into-one

**Comment by zhengruifeng** [comment]
ok， I use std::abs instead

**Comment by tqchen** [comment]
please combine your commits into one using steps as in https://xgboost.readthedocs.org/en/latest/dev-guide/contribute.html#how-to-combine-multiple-commits-into-one

**Comment by zhengruifeng** [comment]
There may be something wrong with my combination operation. What about closing this PR? and I will send another PR?

**Comment by tqchen** [comment]
yes, let us do that then

**Pull request closed without merging**